### DISCUSSION TITLE
开始游戏一直在下载更新

### DISCUSSION THREAD

**Discussion by YansHJ** [Melledy/Grasscutter#1115] [Q&A]
使用的远端服务端，本地主机客户端形式，今天本来官服要更新，本地已经提前下载预载更新包并解压，但是连接服务器时显示还需要下载10GB资源，请问这个该如何解决呢
![_KL4NR~L$OY~7QGAAJFI}JS](https://user-images.githubusercontent.com/56545309/171070161-552057e6-1384-4c09-8c2a-44375cd5684f.png)

**Comment by YansHJ** [comment]
The remote server used is in the form of a local host client. Today, the official server was supposed to be updated. The local has downloaded and decompressed the preloaded update package in advance, but it still needs to download 10GB resources when connecting to the server. How can I solve this problem

**Comment by YansHJ** [comment]
目前将服务端更新至v1.1.2-dev，并且开启对应的端口以及修改了config.json后，没有再出现这个问题，但是新的问题是成功登陆后，点击进入游戏会白屏一阵子，之后弹出提示“网络响应超时，是否重新登陆”，无法进入游戏

**Reply by zhaodice** [reply]
白屏的时候控制台有输出什么?

**Reply by YansHJ** [reply]
![QQ截图20220531132924](https://user-images.githubusercontent.com/56545309/171134978-070484d2-7a93-4871-80a6-86665c10a89e.png)
![QQ截图20220531133155](https://user-images.githubusercontent.com/56545309/171135334-1ac2857a-6f39-4ab6-810f-49fe0eefa226.png)
![`(F~FAO%T1QT(_ E6LEXK T](https://user-images.githubusercontent.com/56545309/171135378-28a1b416-f427-4b79-b05f-ca23141095fe.png)
![B05(6Y% 9APOV}(WH`07WHB](https://user-images.githubusercontent.com/56545309/171135398-c29dee3c-64b1-4e62-b4fd-5d31935453dd.png)

**Reply by aSatone** [reply]
请问这个白屏问题是怎么解决的呢

**Reply by yoimiya43** [reply]
白屏问题+1

**Reply by YansHJ** [reply]
> 请问这个白屏问题是怎么解决的呢

下载最新的2.7版本

**Reply by YansHJ** [reply]
> 请问一下这个问题怎么解决呀 我目前也是这样

下载最新的2.7版本

**Reply by YansHJ** [reply]
> 白屏问题+1

下载最新的2.7版本

**Comment by zhaodice** [comment]
哈，我记得上次没处理好这个游戏的握手包就会这样，有可能是握手包需要更新了~

**Reply by YansHJ** [reply]
> 哈，我记得上次没处理好这个游戏的握手包就会这样，有可能是握手包需要更新了~

你的服务端和安卓端恢复了吗

**Reply by zhaodice** [reply]
> > 哈，我记得上次没处理好这个游戏的握手包就会这样，有可能是握手包需要更新了~
> 
> 你的服务端和安卓端恢复了吗

安卓端连主界面的登录都过不了，怀疑是ssl欺骗失败，电脑端要等我回去了才能试

**Reply by YansHJ** [reply]
> > > 哈，我记得上次没处理好这个游戏的握手包就会这样，有可能是握手包需要更新了~
> > 
> > 
> > 你的服务端和安卓端恢复了吗
> 
> 安卓端连主界面的登录都过不了，怀疑是ssl欺骗失败，电脑端要等我回去了才能试

有点难受，昨天搞到凌晨终于可以进去了，还没玩呢今天就这样了

**Reply by zhaodice** [reply]
> > > > 哈，我记得上次没处理好这个游戏的握手包就会这样，有可能是握手包需要更新了~
> > > 
> > > 
> > > 你的服务端和安卓端恢复了吗
> > 
> > 
> > 安卓端连主界面的登录都过不了，怀疑是ssl欺骗失败，电脑端要等我回去了才能试
> 
> 有点难受，昨天搞到凌晨终于可以进去了，还没玩呢今天就这样了

你这个情况，我几乎可以100%确定，最后的官方更新把握手包改了，这个你有条件抓包吗?登官方服务器，抓一下进大门的包

**Comment by zhaodice** [comment]
你是电脑端，有条件用shark来抓包吗?(官方服务器登录)康康握手包

**Reply by YansHJ** [reply]
> 你是电脑端，有条件用shark来抓包吗?(官方服务器登录)康康握手包

我可能不太会这方面的操作

**Comment by zhaodice** [comment]
那等我回去抓一下包试试看吧，说实话我挺担心官方搞了什么骚东西，如果有啥特殊加密我也没什么办法

**Reply by YansHJ** [reply]
> 那等我回去抓一下包试试看吧，说实话我挺担心官方搞了什么骚东西，如果有啥特殊加密我也没什么办法

嗯嗯，希望不是，但刚好今天官服更新，然后就登陆不上了，估计还是有关联的

**Reply by YansHJ** [reply]
> 那等我回去抓一下包试试看吧，说实话我挺担心官方搞了什么骚东西，如果有啥特殊加密我也没什么办法

哈喽，打扰一下，你这边有什么结果吗

**Reply by zhaodice** [reply]
> > 那等我回去抓一下包试试看吧，说实话我挺担心官方搞了什么骚东西，如果有啥特殊加密我也没什么办法
> 
> 哈喽，打扰一下，你这边有什么结果吗

电脑端还没弄，但我用手机抓了一下包（官服的），发现握手包没变，但手机端没办法连私服，被ssl证书挡了，所以用电脑再试试看

**Reply by YansHJ** [reply]
> > > 那等我回去抓一下包试试看吧，说实话我挺担心官方搞了什么骚东西，如果有啥特殊加密我也没什么办法
> > 
> > 
> > 哈喽，打扰一下，你这边有什么结果吗
> 
> 电脑端还没弄，但我用手机抓了一下包（官服的），发现握手包没变，但手机端没办法连私服，被ssl证书挡了，所以用电脑再试试看

我这边不太好弄，我早上才换成了v1.1.2-dev已经出问题了，不太清楚这个版本正常情况下是什么样子的

**Comment by zhaodice** [comment]
@YansHJ 我试了，电脑端可以进去，没有出现超时

**Reply by YansHJ** [reply]
> @YansHJ 我试了，电脑端可以进去，没有出现超时

2.7是那个版本？是那个分支吗？我这边拉的是v1.1.2-dev ，你那边的服务端是那个版本的

**Reply by zhaodice** [reply]
> > @YansHJ 我试了，电脑端可以进去，没有出现超时
> 
> 2.7是那个版本？是那个分支吗？我这边拉的是v1.1.2-dev ，你那边的服务端是那个版本的

https://github.com/Grasscutters/Grasscutter/actions/runs/2414070112

**Reply by YansHJ** [reply]
谢谢，我试一下

**Comment by zhaodice** [comment]
![image](https://user-images.githubusercontent.com/63996691/171199243-a0df9885-d19a-43df-ad99-df0fd08647dc.png)
电脑端一切正常，2.7

**Comment by zhaodice** [comment]
嗯...不过还是有问题，角色不能攻击

**Reply by YansHJ** [reply]
是个很奇怪的问题，尝试重启游戏吗

**Reply by YansHJ** [reply]
对了，通过你刚才给我的连接，我只找到了一个jar包，并没有其他的相关文件，我应该怎么通过这个jar去构建服务器，直接java -jar 去启动吗

**Reply by zhaodice** [reply]
> 对了，通过你刚才给我的连接，我只找到了一个jar包，并没有其他的相关文件，我应该怎么通过这个jar去构建服务器，直接java -jar 去启动吗

裆燃啦，这是已经编译好的可执行java程序啊

**Reply by YansHJ** [reply]
好像还需要单独下载一些文件比如resources之类的，我正在尝试

**Reply by YansHJ** [reply]
昨天尝试之后我自己可以进去游戏了，但是我有一些朋友进不去，他们会卡在登录界面，然后控制台会卡在这一步，你有遇到过这种情况吗
![3~57W VO6S1W6{~UIM@2GWG](https://user-images.githubusercontent.com/56545309/171327590-df312055-a0a8-42d7-80f9-8de4a0aa0a56.png)

**Reply by YansHJ** [reply]
这个请求，query_region_list是有什么作用呢

**Reply by YansHJ** [reply]
解决了，网的问题，移动的网不行

**Comment by zhaodice** [comment]
我重新下载了resource文件夹，发现可以攻击了，目前还算正常，希望以后服务器越来越完善

**Reply by dint0** [reply]
请问在哪下的，能否给个地址呢，恕我愚笨没找到

**Reply by YansHJ** [reply]
> 请问在哪下的，能否给个地址呢，恕我愚笨没找到

在上面WIki里，有个Run相关的，里面有resource文件

**Reply by YansHJ** [reply]
> 我重新下载了resource文件夹，发现可以攻击了，目前还算正常，希望以后服务器越来越完善

我这边已经可以正常进游戏了，但还是发现有些问题，比如不能邀请2P进副本，部分地方的怪不刷新等等

**Reply by dint0** [reply]
再次请问，游戏进行到选择人物输入昵称时，点击确认就会断开链接能否解决呢，叨扰了

**Reply by zhaodice** [reply]
> 再次请问，游戏进行到选择人物输入昵称时，点击确认就会断开链接能否解决呢，叨扰了

需要dev版本
